vars/commonPipelineEnvironment.groovy

```diff
@@ -47,6 +47,8 @@ class commonPipelineEnvironment implements Serializable {
 
     String abapAddonDescriptor
 
+    String iFlowMplStatus
```

Comment:
@OliverNocon: any concerns wrt update of the CPE here?

Reply:
this will likely bring us trouble when interacting with the binary.

I suggest to use the custom values of the CPE instead of adding another parameter.
There we have getters and setters and `piperExecuteBin` handles it properly

Reply:
I guess @OliverNocon suggests to use

```
private Map valueMap = [:]

    void setValue(String property, value) {
        valueMap[property] = value
    }

    def getValue(String property) {
        return valueMap.get(property)
    }

```

instead of adding new properties.

Reply:
Hi Oliver/Marc,

i don't see an example implementation in this way in other cmd samples. can you provide some example for the same.

Best Regards,
mayur

Reply:
I had a look into `vars/commonPipelineEnvironment.groovy`, method `readFromDisk`. The valueMap is populated [here](https://github.com/SAP/jenkins-library/blob/master/vars/commonPipelineEnvironment.groovy#L231). So the corresponding value needs to be persisted in folder `custom`. As far as I can see there is one [example](https://github.com/SAP/jenkins-library/blob/master/resources/metadata/whitesource.yaml#L408) for that in the sources. From that the generator produces [this](https://github.com/SAP/jenkins-library/blob/master/cmd/whitesourceExecuteScan_generated.go#L59), which gets populated [here](https://github.com/SAP/jenkins-library/blob/master/cmd/whitesourceExecuteScan.go#L634).

@mayurmohan Never use this myself. Hope this helps .. I more the less reverse-engineered this.
@OliverNocon : something to amend?

Reply:
CPE concept i had changed, let me know changes are fine?